malariagen_data/anopheles.py

```diff
@@ -6358,20 +6361,52 @@ def plot_samples_interactive_map(
             df_location_aggs, on=location_composite_key, validate="one_to_one"
         )
 
-        debug("create a map")
-        if basemap is None:
-            basemap = ipyleaflet.basemaps.Esri.WorldImagery
+        debug("create basemap")
+        basemap_abbrevs = {
+            "mapnik": ipyleaflet.basemaps.OpenStreetMap.Mapnik,
+            "natgeoworldmap": ipyleaflet.basemaps.Esri.NatGeoWorldMap,
+            "opentopomap": ipyleaflet.basemaps.OpenTopoMap,
+            "positron": ipyleaflet.basemaps.CartoDB.Positron,
+            "satellite": ipyleaflet.basemaps.Gaode.Satellite,
+            "terrain": ipyleaflet.basemaps.Stamen.Terrain,
+            "watercolor": ipyleaflet.basemaps.Stamen.Watercolor,
+            "worldimagery": ipyleaflet.basemaps.Esri.WorldImagery,
+            "worldstreetmap": ipyleaflet.basemaps.Esri.WorldStreetMap,
+            "worldtopomap": ipyleaflet.basemaps.Esri.WorldTopoMap,
+        }
+
+        # Get basemap_provider
+        if type(basemap) == str:
+            basemap_str = basemap.lower()
+            if basemap_str not in basemap_abbrevs:
+                raise ValueError("Basemap abbreviation not recognised:", basemap_str)
+            basemap_provider = basemap_abbrevs[basemap_str]
+        elif basemap is None:
+            basemap_provider = ipyleaflet.basemaps.Esri.WorldImagery
+        else:
+            basemap_provider = basemap
+
+        # Get basemap_abbrev_selected
+        basemap_abbrev_selected = None
+        if basemap_provider in basemap_abbrevs.values():
+            basemap_abbrev_selected = list(basemap_abbrevs.keys())[
+                list(basemap_abbrevs.values()).index(basemap_provider)
+            ]
+
+        debug("create samples map")
         samples_map = ipyleaflet.Map(
             center=center,
             zoom=zoom,
-            basemap=basemap,
+            basemap=basemap_provider,
         )
         scale_control = ipyleaflet.ScaleControl(position="bottomleft")
         samples_map.add_control(scale_control)
         # make the map a bit taller than the default
         samples_map.layout.height = "500px"
 
         debug("add markers")
+        # TODO: mosquito & mosquito-net icons were added to Font Awesome in 4.10. (ipyleaflet has 4.7.0.)
+        icon = ipyleaflet.AwesomeIcon(name="circle")
```

Comment:
Maybe could add an `icon` parameter to the function, default to "circle"?

Reply:
Will do.
Could alternatively just default to `None`, which I suspect then just defaults to circle (or something identical) through `ipyleaflet.Marker`.